common/parser.hpp

```diff
@@ -194,50 +197,37 @@ struct commands_xml
     std::map<std::string, custom_formatter> custom_formatters;
 };
 
-inline std::string hexify(unsigned char n)
+// While working from files, convert the file content into recognized raw format
+inline void preprocess_data_payload(std::vector<std::string>& params)
 {
-    std::string res;
+    if (!params.size())
+        return;
 
-    do
-    {
-        res += "0123456789ABCDEF"[n % 16];
-        n >>= 4;
-    } while (n);
+    //The last parameter provided by the user can represent a fully-qualified file path
+    std::vector<std::string> modified_params = params;
 
-    std::reverse(res.begin(), res.end());
+    std::string str_param = params.back();
 
-    if (res.size() == 1)
+    if (regex_match(str_param, std::regex(".*\\.*$")))
     {
-        res.insert(0, "0");
-    }
-
-    return res;
-}
-
-inline std::string to_lower(std::string x)
-{
-    transform(x.begin(), x.end(), x.begin(), tolower);
-    return x;
-}
-
-inline bool string_to_bool(const std::string& x)
-{
-    return (to_lower(x) == "true");
-}
+        std::ifstream file(str_param, std::ios::binary | std::ios::in);
 
-inline unsigned int string_to_hex(std::string str)
-{
-    std::string delimiter = "x";
-
-    auto pos = str.find(delimiter);
-    str.erase(0, pos + delimiter.length());
+        if (file.good())
+        {
+            auto data = std::vector<uint8_t>((std::istreambuf_iterator<char>(file)),
+                std::istreambuf_iterator<char>());
+            modified_params.clear();
```

Comment:
If params is `1 2 3 <filename>`...
This clears it.
The next line resizes to fit the data.
The one after assigns the data to `? ? ? hex hex hex ...`.
The first 3 bytes remain 0s...

Reply:
Done